bootstrap-app/src/cmr/bootstrap/data/bulk_index.clj

```diff
@@ -119,6 +113,39 @@
             gran-count
             provider-id)))
 
+(defn- index-variables-by-provider
+  "Bulk index variables for the given provider."
+  [system provider]
+  (info "Indexing variables for provider" (:provider-id provider))
+  (let [db (helper/get-metadata-db-db system)
+        concept-batches (db/find-concepts-in-batches
+                         db provider
+                         {:concept-type :variable}
+                         (:db-batch-size system))
+        num-variables (index/bulk-index
+                       {:system (helper/get-indexer system)}
+                       concept-batches
+                       {})]
+    (info (format "Indexing of %s variables completed." num-variables))
+    num-variables))
+
+(defn index-variables
+  "Bulk index variables for the given provider-id."
+  [system provider-id]
+  (->> provider-id
+       (helper/get-provider system)
+       (index-variables-by-provider system)))
+
+(defn index-all-variables
+  "Bulk index all CMR variables."
+  [system]
+  (info "Indexing all variables")
+  (->> system
+       (helper/get-providers)
+       (map (partial index-variables-by-provider system))
+       vec)
```

Comment:
Okay; I prefer the FP/data-flow cleanliness of what's there, but I can change it to `doseq`.